### PR TITLE
static end point allocation

### PR DESCRIPTION
**Pull request by arnop2**
Add patches that suppress the memory allocation/free in openamp lib.
with these patches application have to declare the endpoint ( dynamic or static allocation) and provide
the structure address as parameter of the  rpmsg_create_ept function.
On the other side, when a new service announcement is received, no more endpoint is created. 
Instead, a user callback is created to provide service name and remote address.  application can:
- create the associated endpoint in callback ( calling rpmsg_create_ept )
- save the name and the remote address for future use
- Ignore the service as not supported. in this case the service request is dropped.

### CODE REVIEW

File: lib/rpmsg/rpmsg.c [hunk 1]
```diff
@@ -271,7 +269,7 @@ struct rpmsg_endpoint *rpmsg_create_ept(struct rpmsg_device *rdev,
 			goto ns_err;
 	}
 
-	return ept;
+	return 0;
 
 ns_err:
 	rpmsg_unregister_endpoint(ept);
```
Comment (wjliang) [inline lib/rpmsg/rpmsg.c]: In  case of error, shall we also need to unset the endpoint address?
Or maybe we can assume registering endpoint always succeed. as it just add to  link list

File: lib/rpmsg/rpmsg_virtio.c [hunk 2]
```diff
@@ -403,35 +403,25 @@ static void rpmsg_virtio_ns_callback(struct rpmsg_endpoint *ept, void *data,
...
-			_ept->addr = RPMSG_ADDR_ANY;
-			_ept->dest_addr = ns_msg->addr;
-			status = rpmsg_register_endpoint(rdev, _ept);
-			if (status < 0) {
-				metal_free_memory(ept);
-				return;
-			}
-			if (rdev->new_endpoint_cb)
-				rdev->new_endpoint_cb(_ept);
-		} else {
```
[the remark this answers is not in the document]
Comment (wjliang) [reply lib/rpmsg/rpmsg_virtio.c]: yes, in case endpoint is found, and we need to set the dest_addr

File: lib/rpmsg/rpmsg.c [hunk 3]
```diff
@@ -197,7 +197,7 @@ struct rpmsg_endpoint *rpmsg_get_endpoint(struct rpmsg_device *rdev,
 			return ept;
 		if (addr == RPMSG_ADDR_ANY &&
 		    ept->dest_addr == RPMSG_ADDR_ANY &&
-		    name_match)
+		    name && name_match)
```
Comment (wjliang) [inline lib/rpmsg/rpmsg.c]: why name is required? as when setting name_match, it already checks name

File: lib/rpmsg/rpmsg_virtio.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (arnop2) [reply lib/rpmsg/rpmsg_virtio.c]: yes, missing i will add it, thanks

[the remark this answers is not in the document]
Comment (arnop2) [reply lib/rpmsg/rpmsg_virtio.c]: Sorry forget my previous comment, i answered too quickly . We look for the _ept based on the remote_addr, that means that the dest_addr is already set.  
The use case is a static endpoint registration that specify the des_addr. I can add a control on the service name to ensure that the new service name match with the registered one. But how should we treat the error case?

**Comment by wjliang** [comment]
close this as V2 is there

**Pull request closed without merging**